packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -183,6 +199,34 @@ export function CategoryAutocomplete({
     [defaultCategoryGroups, categoryGroups, showSplitOption],
   );
 
+  const filterSuggestions = useCallback(
+    (
+      suggestions: CategoryAutocompleteItem[],
+      value: string,
+    ): CategoryAutocompleteItem[] => {
+      return suggestions
+        .filter(suggestion => {
+          const suggestionAsItem = suggestion as CategoryAutocompleteItem; // No need for casting to Suggestion here
+          return (
+            suggestionAsItem.id === 'split' ||
+            suggestionAsItem.group?.name
+              .toLowerCase()
+              .includes(value.toLowerCase()) ||
+            (suggestionAsItem.group?.name + ' ' + suggestionAsItem.name)
+              .toLowerCase()
+              .includes(value.toLowerCase()) ||
+            defaultFilterSuggestion(suggestionAsItem, value)
+          );
+        })
+        .toSorted(
+          (a, b) =>
+            customSort(a, value.toLowerCase()) -
+            customSort(b, value.toLowerCase()),
+        );
+    },
+    [], // Dependencies array
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
    [],
```